src/payment/PaymentPage.jsx

```diff
@@ -51,44 +48,50 @@ class PaymentPage extends React.Component {
               defaultMessage="If you attempted to make a purchase, you have not been charged. Return to your {actionLinkOne} to try again, or {actionLinkTwo}."
               description="The message displayed when there is no basket. Action links will redirect to dashboard or support page"
               values={{
-              actionLinkOne: (
-                <Hyperlink destination={dashboardURL}>
-                  <FormattedMessage
-                    id="payment.empty.basket.dashboardURL"
-                    defaultMessage="dashboard"
-                    description="The message displayed on the redirect to dashboard link"
-                  />
-                </Hyperlink>
-              ),
-              actionLinkTwo: (
-                <Hyperlink destination={supportURL}>
-                  <FormattedMessage
-                    id="payment.empty.basket.supportURL"
-                    defaultMessage="contact edX E-commerce Support"
-                    description="The message displayed on the redirect to support page link"
-                  />
-                </Hyperlink>
-              ),
-            }}
+                actionLinkOne: (
+                  <Hyperlink destination={dashboardURL}>
+                    <FormattedMessage
+                      id="payment.empty.basket.dashboardURL"
+                      defaultMessage="dashboard"
+                      description="The message displayed on the redirect to dashboard link"
+                    />
+                  </Hyperlink>
+                ),
+                actionLinkTwo: (
+                  <Hyperlink destination={supportURL}>
+                    <FormattedMessage
+                      id="payment.empty.basket.supportURL"
+                      defaultMessage="contact edX E-commerce Support"
+                      description="The message displayed on the redirect to support page link"
+                    />
+                  </Hyperlink>
+                ),
+              }}
             />
           </p>
         </div>
       </div>
     );
   }
 
-  renderLoading() {
+  /**
+   * If we're going to be redirecting to another page instead of showing the user the interface,
+   * show a minimal spinner while the redirect is happening.  In other cases we want to show the
+   * page skeleton, but in this case that would be misleading.
+   */
+  renderRedirectSpinner() {
     return (
       <PageLoading srMessage={this.props.intl.formatMessage(messages['payment.loading.payment'])} />
     );
   }
 
+  /**
+   * We show the basket view for all cases except: 1) an empty basket, and 2) when we're going to
+   * perform a redirect.  That means that sometimes it's used during loading, in which case it shows
+   * a "skeleton" view of the left-hand side of the interface until the actual content arrives.
+   */
```

Comment:
I think this is enough as is, but it did make me think, maybe SkeletonView (LoadingView?) should be its own thing?  It's fine either way though.

Reply:
Thought about that - the problem/reason I didn't change it is that the left-hand side may be a skeleton, but the right-hand side of renderBasket() is all the real content.  I think it'd almost be just as confusing if renderSkeleton() and renderBasket() were both rendering the real forms on the right-hand side. :)  Either way the method name isn't quite describing what's going on in it, imho...